### ISSUE TITLE
Multiple ssh keys loaded in mgit, wrong one sent to server

### ISSUE THREAD

**Issue by jcflack** [maks/MGit#490]
Hi,

mgit 1.5.5.1 on android 10 from f-droid.

The first repository I cloned went perfectly. I installed the ssh keypair, gave the URL to the remote server, the clone happened, everything worked.

So I wanted to clone another repo from a different server. I generated a new keypair in mgit (4096-bit rsa), copied the public side to `authorized_keys` on my server, tried the clone, and got 'authentication failed'.

Running my server's sshd with `-d -d -d`, I can see these messages:

```
debug2: input_userauth_request: try method publickey [preauth]
debug3: userauth_pubkey: have signature for RSA SHA256:QpTNemwwc24CiGg...BGJAI [preauth]
...
debug2: user_key_allowed: check options: 'no-port-forwarding,no-agent-forwarding,command="git-shell -c \\"$SSH_ORIGINAL_COMMAND\\"" ssh-rsa AAAAB3NzaC1y...w== sgit
debug2: key not found
```

Checking the key fingerprint in that `userauth_pubkey` line, that's the fingerprint of the first keypair I added in mgit ... the correct one for the first server I cloned the first repo from.

It isn't the fingerprint of the new keypair I just generated. mgit never sends that one. It just offers the wrong one, fails authentication, and gives up.

Maybe I have just overlooked something in the UI where I say which key to use? There doesn't seem to be any way in the clone dialog to specify.

How should I use mgit with more than one known keypair?

**Comment by jcflack** [comment]
My guess: the problem is [this](https://github.com/maks/MGit/blob/master/app/src/main/java/me/sheimi/sgit/AndroidJschCredentialsProvider.java#L36).

**Comment by maks** [comment]
Hopefully Fixed in #492
